### PR TITLE
dialects: (stablehlo) add stablehlo.case

### PR DESCRIPTION
**Pull request by efferifick**
Adds stablehlo.case

### CODE REVIEW

**Comment by efferifick** [comment]
Noticed that multiple returns are printed differently in xdsl.

```
// CHECK: %0, %1 = "stablehlo.case"(%index) ({
%0:2 = "stablehlo.case"(%index) ({
```

File: xdsl/dialects/stablehlo.py [hunk 1]
```diff
@@ -176,6 +186,42 @@ def __init__(
...
+    name = "stablehlo.case"
+    index = operand_def(SI32TensorType)
+    branches: VarRegion = var_region_def("single_block")
+    _results = var_result_def(AnyTensorType | TokenType)
+
+    def __init__(
+        self,
+        index: SSAValue,
+        branches: list[Region],
+        results: list[AnyTensorType | TokenType],
```
Comment (superlopuh) [inline xdsl/dialects/stablehlo.py:218]:
```suggestion
        branches: Sequence[Region],
        result_types: Sequence[AnyTensorType | TokenType],
```

File: xdsl/dialects/stablehlo.py [hunk 2]
```diff
@@ -176,6 +186,42 @@ def __init__(
...
+
+    selected_branch = branches[index] if 0 <= index < size(branches).
+    selected_branch = branches[-1] otherwise.
+
+    https://github.com/openxla/stablehlo/blob/main/docs/spec.md#case
+    """
+
+    name = "stablehlo.case"
+    index = operand_def(SI32TensorType)
+    branches: VarRegion = var_region_def("single_block")
```
Comment (superlopuh) [inline xdsl/dialects/stablehlo.py:211]:
```suggestion
    branches = var_region_def("single_block")
```

**Review by superlopuh — APPROVED** [review]
Looks good modulo suggested changes

File: xdsl/dialects/stablehlo.py (hunk 2 above)
Comment (efferifick) [reply xdsl/dialects/stablehlo.py]: https://github.com/xdslproject/xdsl/pull/3095/commits/0972f49d052b3344e0b6d68f464989cfcd18c176

File: xdsl/dialects/stablehlo.py (hunk 1 above)
Comment (efferifick) [reply xdsl/dialects/stablehlo.py]: https://github.com/xdslproject/xdsl/pull/3095/commits/0972f49d052b3344e0b6d68f464989cfcd18c176

**Comment by efferifick** [comment]
@superlopuh fixed merge conflicts

**Review by superlopuh — APPROVED** [review]

**Pull request merged**